ibu-imager/seedcleanup/seedcleanup.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seedcleanup
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// Default location for etcdStaticPodFile
+	etcdStaticPodFile      = "/etc/kubernetes/manifests/etcd-pod.yaml"
+	etcdStaticPodContainer = "etcd"
+
+	containerListDoneFile = "/var/tmp/container_list.done"
+
+	// Default location for backing up seed cluster certificates
+	backupCertsDir = "/var/tmp/backupCertsDir"
+)
+
+// SeedCleaner handles cleanup operations after seeding, removing temporary files
+// and executing additional cleanup steps as needed.
+type SeedCleaner struct {
+	log                  *logrus.Logger
+	ops                  ops.Ops
+	backupDir            string
+	containerRegistry    string
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+// NewSeedCleaner is a constructor function for SeedCleaner
+func NewSeedCleaner(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedCleaner {
+
+	return &SeedCleaner{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after seeding an SNO cluster.
+func (s *SeedCleaner) CleanupSeedCluster() error {
+	s.log.Println("Cleaning up seed cluster")
+
+	// cleanup backup dir
+	if err := s.cleanupBackupDir(); err != nil {
+		return err
+	}
+
+	// cleanup preparation scripts and services
+	if err := s.cleanupConfigurationFiles(); err != nil {
+		return err
+	}
+
+	// restore certificates in etcd
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring certificates in etcd via recert tool.")
+	} else {
+		if err := s.restoreCertificates(); err != nil {
+			return err
+		}
+	}
+
+	// restore cluster services
+	if err := s.restoreCluster(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SeedCleaner) cleanupBackupDir() error {
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", s.backupDir, err)
+	}
+
+	if err := os.Remove(containerListDoneFile); err != nil {
```

Comment:
I assume we'll want to cleanup all "done" files? Why are we cleaning the containers list done file here in particular/ We should put all done files in the same directory so they can be easily cleaned up

Reply:
couldn't agree more, added #91 for that